### ISSUE TITLE
3.0.0 stuck at SplashScreen 

### ISSUE THREAD

**Issue by mgrbose** [ionic-team/cordova-plugin-ionic#44]
Hello, I am getting below error (in logcat) when new version is available and stuck at SplashScreen. SplashScreen is not hidden in this case.  I am testing on Android device. Thanks..

01-17 19:59:27.804 19854 19915 I IONIC.DEPLOY.EXTRACT: Progress: 96%
**01-17 19:59:27.806 19854 19915 I IONIC.DEPLOY.UNZIP_STEP: Exception: Attempt to invoke virtual method 'void org.apache.cordova.CallbackContext.success(java.lang.String)' on a null object reference**
01-17 19:59:27.813 19854 19915 I IONIC.DEPLOY.REMOVE: Removed www.zip

ionic info
cli packages: (/usr/local/lib/node_modules)
    ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:
    cordova (Cordova CLI) : 7.0.1 

local packages:
    ionic/app-scripts : 3.1.2
    Cordova Platforms  : android 6.3.0
    Ionic Framework    : ionic-angular 3.9.2

System:
    Android SDK Tools : 26.1.1
    ios-deploy        : 1.9.0 
    Node              : v8.3.0
    npm               : 5.3.0 
    OS                : macOS Sierra
    Xcode             : Xcode 9.2 Build version 9C40b 

Environment Variables:
    ANDROID_HOME : /Users/***/Library/Android/sdk

Misc:
    backend : pro

**Comment by vladretca** [comment]
I'm experiencing the same issue on a real device

**Comment by sfaizanh** [comment]
Same issue on iOS
```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 7.1.0 

local packages:

    @ionic/app-scripts : 3.1.7
    Cordova Platforms  : ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    ios-deploy : 1.9.2 
    ios-sim    : 5.0.6 
    Node       : v8.9.4
    npm        : 5.6.0 
    OS         : macOS High Sierra
    Xcode      : Xcode 9.2 Build version 9C40b 

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro
```

**Comment by KesselringWebmaster** [comment]
I also have the same issue at android. 
I am getting the error when android downloads the update and then it hangs on the splash screen. 
This is the link on the screenshot where I also display console output of Android Studio Debugger. 
https://www.dropbox.com/s/2uyci6j23wgsoyd/bug.png?dl=0

ionic info
cli packages: (C:\Users\Kesselring\AppData\Roaming\npm\node_modules)
    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1
global packages:
    cordova (Cordova CLI) : 8.0.0
local packages:
    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 6.3.0
    Ionic Framework    : ionic-angular 3.9.2
System:
    Android SDK Tools : 26.0.1
    Node              : v8.9.3
    npm               : 5.3.0
    OS                : Windows 7
Environment Variables:
    ANDROID_HOME : D:\Android\Sdk
Misc:
    backend : pro

**Comment by wstrinz** [comment]
Same issue, same error message in logcat.

```
> ionic info

cli packages: (/Users/willstrinz/.nvm/versions/node/v9.2.0/lib/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.1.5
    Cordova Platforms  : android 6.3.0 ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 25.2.5
    ios-deploy        : 1.9.2
    Node              : v9.2.0
    npm               : 5.5.1
    OS                : macOS High Sierra
    Xcode             : Xcode 9.2 Build version 9C40b

Environment Variables:

    ANDROID_HOME : /usr/local/Cellar/android-sdk/24.4.1_1/

Misc:

    backend : pro

```

**Comment by KesselringWebmaster** [comment]
Hey,
I have an idea. What if this bug happens due to the fact, that android version was not build with "--prod" flag.
I just realized that when I tried to deploy live code update, I used a release apk that was built for old ionic 1 way:
"ionic cordova build android --release"
However, what the official doc says: the --prod flag has to be added. 
The full command should be "ionic cordova build android --release --prod"

let me know if it helped you, I have not tried this yet though

**Comment by tobbbe** [comment]
@KesselringWebmaster tried it now but didnt work unfortunately

**Comment by none100p** [comment]
Same issue here, tested on a real device, and emulator.
Made tests only for android, for now.
ionic info

cli packages: (/home/none100p/.nvm/versions/node/v6.11.5/lib/node_modules)

    @ionic/cli-plugin-proxy : 1.5.6
    @ionic/cli-utils        : 1.19.1
    ionic (Ionic CLI)       : 3.19.1

global packages:

    cordova (Cordova CLI) : 8.0.0 

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 6.4.0 browser 5.0.3
    Ionic Framework    : ionic-angular 3.9.2

System:

    Node : v6.11.5
    npm  : 3.10.10 
    OS   : Linux 4.4

Misc:

    backend : pro

**Comment by mhilland** [comment]
I'm also experiencing the very same issue. Stuck at splash screen with the error `IONIC.DEPLOY.UNZIP_STEP: Exception: Attempt to invoke virtual method 'void org.apache.cordova.CallbackContext.success(java.lang.String)' on a null object reference`

I have tried downgrading to 2.0.4 and I get the same thing. It works fine on iOS.

Using cordova-android 5.2.2

**Comment by Fuiste** [comment]
This should be fixed in version 3.1.0 of the plugin, which was just released.  I've tested on iOS and android, and both appear to be working as expected.
